File: Codes.py
## Load the VIIRS image collection
var viirsCollection = ee.ImageCollection('NOAA/VIIRS/001/VNP46A2');
## Define the date range
var startDate = '2022-01-01';
var endDate = '2022-12-30';

## Define the coordinates for the pixel
var point1 = ee.Geometry.Point(147.3250000, -42.89583335);
var point2 = ee.Geometry.Point(147.3291667, -42.9000000);
var pixelGeometry = ee.Geometry.Rectangle(point1, point2);

## Filter the image collection based on the date range and pixel geometry
var filteredCollection = viirsCollection.filterDate(startDate, endDate)
  .filterBounds(pixelGeometry);

## Select the band of interest
var bandName = 'DNB_BRDF_Corrected_NTL';
var imageWithBand = filteredCollection.select(bandName);

## Define a function to extract the date from the image metadata
var extractDate = function(image) {
  var dateString = ee.String(image.get('system:index'));
  var date = ee.Date.parse('YYYYMMdd', dateString);
  return image.set('date', date);
};

## Map the function over the image collection to extract the date
var imageCollectionWithDate = imageWithBand.map(extractDate);

## Create a daily composite by reducing the collection by date
var dailyComposite = imageCollectionWithDate.reduce(ee.Reducer.first());

File: Codes.js
//Define the date range
var startDate = '2022-01-01';
var endDate = '2022-12-30';

// Define the coordinates for the pixel
//Make sure both numbers in point1 are smaller than point2
var point1 = ee.Geometry.Point(66.7365, 24.6548); //Karachi
var point2 = ee.Geometry.Point(67.5495, 25.2325); 
//var point1 = ee.Geometry.Point(-115.4726, 35.7251); //Las Vegas
//var point2 = ee.Geometry.Point(-114.728, 36.474);
var pixelGeometry = ee.Geometry.Rectangle([point1, point2]);
Map.centerObject(pixelGeometry.centroid(), 10);//Center Map on point1

// Load VIIRS image collection for 2022
var viirsCollection = ee.ImageCollection('NOAA/VIIRS/001/VNP46A2')
  .filterDate(startDate, endDate)
  .select(bandName); // Select the desired band

// Clip the data to the geometry region
var viirsCollection_clipped = viirsCollection.map(function(image) {
  return image.clip(pixelGeometry);
});

// Display the 1 date in the image collection
// Click on the pixels with the inspector tool to see a time series
Map.addLayer(viirsCollection_clipped, {min: 10, max: 150}, 'VIIRS Nighttime Lights - First Date');

// Calculate the average value for the time period
var viirsMean = viirsCollection_clipped.mean();
Map.addLayer(viirsMean, {min: 10, max: 150}, 'VIIRS Nighttime Lights - Average');
